medium-tests/src/test/java/mediumtest/remediation/aicodefix/AiCodeFixMediumTest.java

```diff
@@ -439,6 +445,56 @@ void it_should_mark_the_issue_as_fixable_if_supported(SonarLintTestHarness harne
     assertThat(issue.isAiCodeFixable()).isTrue();
   }
 
+  @SonarLintTest
+  void it_should_throw_too_many_requests(SonarLintTestHarness harness, @TempDir Path baseDir) {
+    var filePath = createFile(baseDir, "pom.xml", XML_SOURCE_CODE_WITH_ISSUE);
+    var fileUri = filePath.toUri();
+    var server = harness.newFakeSonarCloudServer()
+      .withOrganization("organizationKey", organization -> organization
+        .withProject("projectKey",
+          project -> project
+            .withBranch("branchName")
+            .withAiCodeFixSuggestion(suggestion -> suggestion
+              .withId(UUID.fromString("e51b7bbd-72bc-4008-a4f1-d75583f3dc98"))
+              .withExplanation("This is the explanation")
+              .withChange(0, 0, "This is the new code"))))
+      .start();
+
+    server.getMockServer().stubFor(
+      post(urlMatching(".*/fix-suggestions.*"))
+        .willReturn(aResponse().withStatus(HTTP_TOO_MANY_REQUESTS))
+    );
```

Comment:
Instead of mocking the server here, you could use `withResponseCodes` in the builder that already takes care of that